application/helpers/expressions/em_manager_helper.php

```diff
@@ -9662,73 +9675,12 @@ static public function ShowSurveyLogicFile($sid, $gid=NULL, $qid=NULL,$LEMdebugL
                             }
                             $LEM->ProcessString($value, $qid,$qReplacement,1,1,false,false);
                             $value = viewHelper::stripTagsEM($LEM->GetLastPrettyPrintExpression());
-                            if ($LEM->em->HasErrors())
-                            {
+                            if ($LEM->em->HasErrors()) {
                                 ++$errorCount;
                             }
+                            $aWarnings = array_merge($aWarnings,$LEM->em->GetWarnings());
+                            $LEM->em->ResetWarnings();
                         }
-
-                        //~ switch ($key)
```

Comment:
In the future, please do not mix clean-up with new features in one PR. It makes it harder to read the PR in full. I know, it's hard. :)

Reply:
Hard to do new feature without cleaning up a little …

Reply:
Yeah, takes discipline. But clean-up can be done without PR (mostly).

Reply:
Yes … totally right … sorry … (and clean up master => merge in develop directlty too …)

Reply:
:+1: